qiita_db/test/test_metadata_template.py

```diff
@@ -652,6 +775,133 @@ def test_create(self):
                ['2.Sample3', "Value for sample 3"]]
         self.assertEqual(obs, exp)
 
+    def test_create_int_prefix(self):
+        """Creates a new SampleTemplate"""
+        st = SampleTemplate.create(self.metadata_int_pref, self.new_study)
+        # The returned object has the correct id
+        self.assertEqual(st.id, 2)
+
+        # The relevant rows to required_sample_info have been added.
+        obs = self.conn_handler.execute_fetchall(
+            "SELECT * FROM qiita.required_sample_info WHERE study_id=2")
+        # sample_id study_id physical_location has_physical_specimen
+        # has_extracted_data sample_type required_sample_info_status_id
+        # collection_timestamp host_subject_id description
+        exp = [["2.12.Sample1", 2, "location1", True, True, "type1", 1,
```

Comment:
Yes. Following QIIME rules, the sample id can contain numbers, letters and the `.` character. In qiita we are adding `<study_id>.` as prefix to all sample ids. So everything after the first `.` is the "original" sample id. Note that in every other place in Qiita, the sample id is with the study id prefixed.

That construction of the sample ids was already added a while ago (august, september?) here I'm just adding the option to re-upload the templates downloaded from Qiita; but I'm not adding anything new to the sample ids.